### PR TITLE
Cherry pick of #91099: skip unnecessary scheduling attempt when pod has its Finalizers updated

### PR DESCRIPTION
**Pull request by snowplayfire**
/kind bug

/sig scheduling

Cherry pick of #91099 on release-1.16.

@alculquicondor @ahg-g

### CODE REVIEW

**Comment by snowplayfire** [comment]
@kubernetes/patch-release-team

**Comment by alculquicondor** [comment]
/ok-to-test

Avoid pinging release team until change is approved by owners.

Limit the change to Finalizers.

**Comment by alculquicondor** [comment]
Also, first cherrypick into newer versions and work your way down the releases

**Comment by snowplayfire** [comment]
> Also, first cherrypick into newer versions and work your way down the releases

It's my first time to cherrypick, can you explain it in more detail?  

Thanks!

**Comment by alculquicondor** [comment]
This is targeting 1.16. You should target 1.18 first.

File: pkg/scheduler/eventhandlers.go [hunk 1]
```diff
@@ -366,6 +366,11 @@ func (sched *Scheduler) skipPodUpdate(pod *v1.Pod) bool {
 		// Annotations must be excluded for the reasons described in
 		// https://github.com/kubernetes/kubernetes/issues/52914.
 		p.Annotations = nil
+		// Same as above, when annotations are modified with ServerSideApply,
+		// ManagedFields may also change and must be excluded
+		p.ManagedFields = nil
```
Comment (Huang-Wei) [inline pkg/scheduler/eventhandlers.go:371]: Prefer to remove this out. If it's agreed to back this SSA change, the right way is to backport the original PR (#90373) in a separate PR.

Comment (alculquicondor) [reply pkg/scheduler/eventhandlers.go:371]: We probably don't need to backport that one as far back. Not sure how old Server Side Apply is.

Comment (Huang-Wei) [reply pkg/scheduler/eventhandlers.go:371]: SSA was beta in 1.16, but I heard there was some implication in the first beta version which doesn't record which actor modified the fields (or sth. like that), so 🤷‍♂️ not quite sure it's widely adopted back in 1.16.

> We probably don't need to backport that one as far back.

Agree.

**Comment by snowplayfire** [comment]
close the PR and open a new one. #91245

**Pull request closed without merging**